docs/how-tos/deploy-with-istio.md

```diff
@@ -0,0 +1,76 @@
+# Deploy with Istio
+
+## Preface
+
+This guide will explain how to integrate an OPA instance as part of a Kubernetes deployment making use of the [Istio Service Mesh](https://istio.io/latest/about/service-mesh/) and [Envoy external authorization filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto) to authorize all incoming traffic to a pod against OPA policy.
+
+It is intended to follow on from [deploying OPA with Helm](./deploy-with-helm.md).
+
+## Adjust the OPA deployment
+
+To allow the Envoy sidecar to communicate with the OPA istance, it must use and configure the OPA-Envoy plugin, which requires the following changes to the `values.yaml` of the Helm chart that has opa as a dependency:
+
+```yaml
+opa:
+  image:
+    envoy: true
+  config:
+    plugins:
+      envoy_ext_authz_grpc:
+        path: #  Make this the root of your built policy
+```
+
+## Enable Envoy Sidecar injection
+
+The Istio ServiceMesh operates by injecting a sidecar container alongside your service which intercepts all incoming traffic. To enable the injection of this sidecar, the following label must be added to your service's deployment.
+
+```yaml
+spec:
+  template:
+    metadata:
+      labels:
+        sidecar.istio.io/inject: 'true'
+```
+
+## Installing the External Authorization EnvoyFilter into the Envoy instance
+
+An Envoy sidecar checks all incoming traffic against a series of filters, adjusting, allowing or rejecting traffic accordingly. The following configures an external authorization filter that will check all incoming traffic, passing the headers and path (and optionally the body of the request) to your OPA instance configured above.
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: authz-envoyfilter
+spec:
+#   workloadSelector:  # Optionally apply only to a subset of Envoy sidecars in the namespace
+#     labels:
+#       app: reviews
```

Comment:
I think we can leave this out and assume if people want to do more complex envoy stuff they'll go to the envoy docs

Reply:
I think limiting authz to specific pods is a pretty simple non-envoy specific thing that will probably be required if being deployed as part of a Helm chart. It is for DAQ Athena stuff

Reply:
Fair enough, guess it's just something I've never had to do, but I haven't made good use of envoy yet